includes/class-linkbacks-handler.php

```diff
@@ -33,8 +33,74 @@ public static function init() {
 		add_filter( 'get_comment_author_url', array( 'Linkbacks_Handler', 'get_comment_author_url' ), 99, 3 );
 		add_filter( 'get_avatar_comment_types', array( 'Linkbacks_Handler', 'get_avatar_comment_types' ) );
 		add_filter( 'comment_class', array( 'Linkbacks_Handler', 'comment_class' ), 10, 4 );
+
+		// Register Meta Keys
+		self::register_meta();
 	}
 
+	// This is more to lay out the data structure than anything else.
+	public static function register_meta() {
+		$args = array(
+			'sanitize_callback' => 'esc_url_raw',
+			'type' => 'string',
+			'description' => 'Author URL',
+			'single' => true,
+			'show_in_rest' => true
+		);
+		register_meta( 'comment', 'semantic_linkbacks_author_url', $args );
+
+		$args = array(
+			'sanitize_callback' => 'esc_url_raw',
+			'type' => 'string',
+			'description' => 'Avatar URL',
+			'single' => true,
+			'show_in_rest' => true
+		);
+		register_meta( 'comment', 'semantic_linkbacks_avatar', $args );
+
+		$args = array(
+			'sanitize_callback' => 'esc_url_raw',
+			'type' => 'string',
+			'description' => 'Canonical URL',
+			'single' => true,
+			'show_in_rest' => true
+		);
+		register_meta( 'comment', 'semantic_linkbacks_canonical', $args );
+
+		$args = array(
+			'type' => 'string',
+			'description' => 'Linkbacks Type',
+			'single' => true,
+			'show_in_rest' => true
+		);
+		register_meta( 'comment', 'semantic_linkbacks_type', $args );
+	}
+
+
+	/**
+	 * Enhance a Comment before it is Added to the Database
+	 */
+	public static function pre_enhance( $commentdata ) {
+		if ( ! in_array( $commentdata['comment_type'], array( 'webmention', 'pingback', 'trackback' ) ) ) {
+			return $commentdata;
+		}
+		return self::enhance( $commentdata );
+	}
+
+	/** 
+	 * Update an Enhanced Comment
+	 *
+	 */
+	public static function update( $data, $comment, $commentarr) {
+		if ( ! in_array( $commentarr['comment_type'], array( 'webmention', 'pingback', 'trackback' ) ) ) {
+			return $data;
+		}
+
+		$new = self::enhance( $data );
+		$keys = array( 'comment_post_ID', 'comment_content', 'comment_author', 'comment_author_email', 'comment_approved', 'comment_karma', 'comment_author_url', 'comment_date', 'comment_date_gmt', 'comment_type', 'comment_parent', 'user_id', 'comment_agent', 'comment_author_IP' );
+		$new = wp_array_slice_assoc( $new, $keys );
```

Comment:
The array slicing is necessary until 4.7.1, where this will now be done after the filter instead of before. This has already been committed to 4.7.1 due to the fact not having it causes a database error